gst/nnstreamer/tensor_query/nnstreamer_edge_common.h

```diff
@@ -54,6 +54,12 @@ typedef struct {
   nns_edge_raw_data_s data[NNS_EDGE_DATA_LIMIT];
 } nns_edge_data_s;
 
+typedef struct {
```

Comment:
It looks like you need a lock for nns_edge_event_s. (identify and apply 'minimal' critical sections in nnstreamer_edge_common.c as well)

Reply:
Marked todo item. I did not implement lock code in nns-edge yet.